### ISSUE TITLE
Plotter window hangs around in an inoperable state after close on macOS

### ISSUE THREAD

**Issue by sampotter** [pyvista/pyvista#1096]
On MacOS, if I open up IPython and run:

```py
import pyvista as pv; pv.Sphere().plot()
```

The plotter opens up as expected, and I'm able to interact with it. When I try to close the window, the window doesn't disappear but instead hangs (beach balls).

Weirdly, IPython _is_ still interactive, and the call to `pv.Sphere().plot()` actually returns. E.g., after X-ing out the plotter window:

```py
In [6]: pv.Sphere().plot()
Out[6]:
[(1.7773465342683825, 2.4356984953049423, 1.428995981058186),
 (0.0, 0.0, 0.0),
 (0.4815608961128151, -0.6775433723938147, 0.5559083394411702)]
```

In fact, if I call `pv.Sphere().plot()` again, the X'd out window disappears, having been replaced by a new, fully interactive plotter. I can repeat this process ad infinitum.

I'm not sure if this is relevant, but I tried this with the latest VTK 9 install and also after downgrading to VTK 8.1.2. The problem is the same in both cases.

```
--------------------------------------------------------------------------------
  Date: Thu Jan 21 07:58:22 2021 EST

                OS : Darwin
            CPU(s) : 8
           Machine : x86_64
      Architecture : 64bit
               RAM : 16.0 GB
       Environment : IPython
        GPU Vendor : Intel Inc.
      GPU Renderer : Intel Iris Pro OpenGL Engine
       GPU Version : 4.1 INTEL-14.7.11

  Python 3.7.9 (default, Sep  6 2020, 13:20:25)  [Clang 11.0.3
  (clang-1103.0.32.62)]

           pyvista : 0.27.4
               vtk : 8.1.2
             numpy : 1.19.5
           imageio : 2.9.0
           appdirs : 1.4.4
            scooby : 0.5.6
            meshio : 4.3.8
        matplotlib : 3.3.3
         pyvistaqt : 0.3.0
             PyQt5 : 5.15.2
           IPython : 7.19.0
          colorcet : 1.0.0
             scipy : 1.6.0
--------------------------------------------------------------------------------```
```

**Comment by akaszynski** [comment]
I'm seeing this bug in linux, but with `vtk==9.0.1`.  Could you try upgrading `vtk` and see if the issue persists?

**Comment by sampotter** [comment]
Yes, I said in my original post that the bug happens with both VTK 8 and 9.

**Comment by akaszynski** [comment]
Sorry. Helps if I read these issues completely.

Does this occur on vanilla python and just on iPython?  I'm trying to think if it's due to our last release where we cleaned up the window collection.  Try plotting on normal python, then try downgrading to 0.26.0 and see if the issue persists.

**Comment by sampotter** [comment]
No problem at all! Thanks for the quick responses.

**Trying to plot using vanilla Python interpreter**: the problem persists exactly as before.

**Downgrading to `pyvista==0.26.0**: same problem, for both vanilla Python and IPython.

**Comment by akaszynski** [comment]
Interesting.  I have a Mac somewhere and I'll try to reproduce this locally.

**Comment by sayerhs** [comment]
I also observed this issue when using IPython with pyvista (on MacOS), the plot appears and I am able to interact with it. However, if I hit the quit (`q`) key on the plot window the plot doesn't close but instead hangs with a spinning beach ball icon. It appears to be an interaction issue with IPython threads, and I was able to resolve this issue by executing the `%matplotlib` magic command before executing pyvista plot commands. It appears that the fix for matplotlib thread interaction works also for the pyvista plot windows. Hope this helps others.

**Comment by cristobaltapia** [comment]
I am having a similar problem under Rockylinux 8. I can plot correctly, but once I close the window a new, completely black window opens and cannot be closed. I am using IPython and the latest version of PyVista, 0.37.0 with python 3.8. Once the orignal window is closed, I can send new commands to IPython. If I make another plot, then the new window shortly appears and then I get a "Segmentation fault: (core dumped)". The suggestion of @sayerhs of using `%matplotlib` does not work in this case.

**Comment by banesullivan** [comment]
FYI https://gitlab.kitware.com/vtk/vtk/-/issues/18652

@cristobaltapia, your issue sounds similar and related... perhaps post upstream on that issue as well

**Comment by akaszynski** [comment]
I can reproduce this with VTK 9.2.4 on Linux. It's annoying and I'll fix it.

**Comment by Axion004** [comment]
I can reproduce this issue on PyVista 0.38.1, VTK 9.2.5, and Ubuntu 20.04. I think that it is waiting on a fix from a newer version of VTK. The _kill_display() function didn't fix this issue for me when I tried to close the plotter (render) window through Dash.

**Comment by germa89** [comment]
I can reproduce this in MacOS with silicon architecture. Using main branch.

**Comment by gregoryjanes** [comment]
I can reproduce this in MacOS with silicon and on my Rocky Linux 10 machine. 

Is there any update on this issue? I haven't seen many comments or updates regarding this in the past two years.

```
--------------------------------------------------------------------------------
  Date: Tue Nov 11 15:33:42 2025 UTC

                OS : Linux (Rocky Linux 10.0)
            CPU(s) : 24
           Machine : x86_64
      Architecture : 64bit
               RAM : 62.2 GiB
       Environment : IPython
       File system : xfs
        GPU Vendor : NVIDIA Corporation
      GPU Renderer : NVIDIA GeForce RTX 3080 Ti/PCIe/SSE2
       GPU Version : 4.5.0 NVIDIA 580.105.08
     Render Window : vtkXOpenGLRenderWindow
  MathText Support : True

  Python 3.12.9 (main, Aug 14 2025, 00:00:00) [GCC 14.2.1 20250110 (Red Hat
  14.2.1-7)]

           pyvista : 0.46.3
               vtk : 9.5.2
             numpy : 2.3.4
        matplotlib : 3.10.7
            scooby : 0.10.2
             pooch : 1.8.2
            pillow : 12.0.0
           imageio : 2.37.2
             PyQt5 : 5.15.11
           IPython : 9.6.0
             scipy : 1.16.2
      nest_asyncio : 1.6.0
--------------------------------------------------------------------------------
```

**Comment by akaszynski** [comment]
No updates here in PyVista. Issue is tracked by VTK in https://gitlab.kitware.com/vtk/vtk/-/issues/18652

**Comment by banesullivan** [comment]
To work around this, I've been wanting to combine PyVista, Trame, and pywebview to make a bundled and richer plotting window that could have UI controls and better window management... just don't have the time to do this
